python/hsfs/core/data_validation_engine.py

```diff
@@ -59,15 +59,22 @@ def validate(self, feature_group, feature_dataframe):
             exp = exp_res.getExpectation()
             rules_python = []
             for exp_rule in exp.getRules():
+                legal_values = []
+                if exp_rule.getLegalValues() is not None:
+                    for legal_value in exp_rule.getLegalValues():
+                        legal_values.append(legal_value)
```

Comment:
```suggestion
                legal_values = [legal_value for legal_value in exp_rule.getLegalValues()] if exp_rule.getLegalValues() is not None else []
```
I assume you are doing this because it returns a Java list? are you sure the py4j gateway doesn't convert it to a python list anyway if it's not null?

Reply:
java list to python list didn't work so since this code works I prefer to leave it as is

Reply:
ok with me then